src/ngamsUtils/ngamsUtils/ngasXSyncTool.py

```diff
@@ -295,6 +295,19 @@
 
 _option_dict, _option_document = ngasUtilsLib.generate_options_dictionary_and_document(_options)
 __doc__ = _document.format(_option_document)
+sync_thread_group = None
+
+
+def signal_handler(signum, frame):
+    """
+    Signal handler to set exit flag
+    :param signum: Signal numerical value
+    :param frame: ???
+    :return: None
+    """
+    logger.fatal("Signal handler called with signal %d", signum)
+    if sync_thread_group is not None:
+        sync_thread_group.stop()
```

Comment:
It looks like you are not binding this handler to any signal

Reply:
You are correct. They have now been added.